packages/vue-app/template/index.js

```diff
@@ -166,20 +166,21 @@ async function createApp(ssrContext) {
 
   // Plugin execution
   <%= isTest ? '/* eslint-disable camelcase */' : '' %>
-  <% plugins.filter(p => p.mode === 'all').forEach((plugin) => { %>
-  if (typeof <%= plugin.name %> === 'function') await <%= plugin.name %>(app.context, inject)<% }) %>
-
-  <% if (plugins.filter(p => p.mode === 'client').length) { %>
-  if (process.client) {
-    <% plugins.filter(p => p.mode === 'client').forEach((plugin) => { %>
-    if (typeof <%= plugin.name %> === 'function') await <%= plugin.name %>(app.context, inject)<% }) %>
-  }<% } %>
-
-  <% if (plugins.filter(p => p.mode === 'server').length) { %>
-  if (process.server) {
-    <% plugins.filter(p => p.mode === 'server').forEach((plugin) => { %>
-    if (typeof <%= plugin.name %> === 'function') await <%= plugin.name %>(app.context, inject)<% }) %>
-  }<% } %>
+  <% if (plugins.length) { %>
+    <% plugins.forEach((plugin) => { %>
+      <% if (plugin.mode == 'client') { %>
+        if (process.client) {
```

Comment:
This part would be better to be follow eslint rules in compiled template.

Reply:
Can be also combined with next condition

Reply:
@pi0 but if we combine wont it lead to not treeshaking?
@clarkdo ye, i though its run automatically during tests but seems not :)

Reply:
`typeof` guard is to prevent executing plugins when not exporting a function.

Webpack would compile condition to something like `if (false && typeof ...)` which should be eliminated by minifier.

Tree shaking cannot happen because we don't set side-effects field for above imports BTW it is zero overhead as server plugins as aliased to an empty module for client-side.

Reply:
@pi0 I think import of `empty.js` can be tree-shaked after dead code removal, @aldarund you may test it that if empty.js is removed with you changes.

Reply:
True. The server code is not included in the client bundle regardless of using one line or two separate `if`s.

Reply:
And the reason that client code is not tree-shaked is `await` is transpiled into:

```js
if (true) {
  _context2.next = 33;
  break;
}

return plugin(app.context, inject);
```

So the plugin call isn't removed.

Reply:
changed into one statemement and styled for compiled template